kubehub/views/vm_group_view.py

```diff
@@ -21,32 +21,85 @@ def vm_group_list(request):
     return JsonResponse({"vm_group_list": vm_groups})
 
 
+@csrf_exempt
+def get_vm_group_status(request):
+    if request.method == 'POST':
+        try:
+            data = json.loads(request.body)
+            pk = data.get('vm_group_id')
+            instance = VMGroup.objects.get(pk=pk)
+            return JsonResponse(str(instance.status), safe=False)
+        except Exception as e:
+            return JsonResponse({'errors': {f'{type(e).__name__}': [str(e)]}})
+
+
 @csrf_exempt
 def vm_group_add(request):
     if request.method == 'POST':
         data = json.loads(request.body)
-        vmg_list = create_vm_group(data)
         virtual_machine_group = {
             "name": data['name'],
             "user_id": "1",
-            "vms": vmg_list
+            "status": "creating",
+            "vms": [{
+                "name": "creating",
+                "vmid": "0",
+                "ip": "creating",
+                "template_id": "0",
+                "cloud_provider_id": "0"
+            }for _ in range(int(data["number_of_nodes"]))]
         }
         vmgs = VMGroupSerializer(data=virtual_machine_group)
         if vmgs.is_valid():
-            vmgs.save()
+            created_group = vmgs.create(vmgs.validated_data)
+            pk = created_group.id
+            vmg_list = create_vm_group(data)
+            vms_update(
+                pk=pk,
+                vms=vmg_list
+            )
+            status_update(
+                pk=pk,
+                status="running"
+            )
             return JsonResponse({"data": vmgs.validated_data})
         else:
             return JsonResponse({'errors': vmgs.errors})
 
 
+def vms_update(pk, vms):
+    vms_instance = VM.objects.filter(vm_group__id=pk)
+    updated_vms = []
+    for instance, vm in zip(list(vms_instance), vms):
+        vm_serializer = VMSerializer(data=vm, partial=True)
+        if vm_serializer.is_valid():
+            vm = vm_serializer.update(instance, vm_serializer.validated_data)
+            updated_vms.append(model_to_dict(vm))
+        else:
+            print(vm_serializer.errors)
+
+    return updated_vms
+
+
+def status_update(pk, status):
+    instance = VMGroup.objects.get(pk=pk)
+    data = {"status": status}
+    vmgs = VMGroupSerializer(data=data, partial=True)
+    if vmgs.is_valid():
+        vmg = vmgs.update(instance, vmgs.validated_data)
+        return model_to_dict(vmg)
+
+
 @csrf_exempt
 def vm_group_remove(request):
     if request.method == 'POST':
         try:
             data = json.loads(request.body)
             pk = data.get('vm_group_id')
+            status_update(pk, "removing")
```

Comment:
how long it will take for end user to get response? 10 mins?

Reply:
user will get "removing" status immediately, approximately 10 minutes will be needed to delete group from proxmox (can't use multithreading for this coz proxmox can't handle multiple delete request)